controllers/datadogagent/feature/ebpfcheck/feature.go

```diff
@@ -0,0 +1,131 @@
+package ebpfcheck
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/component/agent"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.EBPFCheckIDType, buildEBPFCheckFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildEBPFCheckFeature(options *feature.Options) feature.Feature {
+	ebpfCheckFeat := &ebpfCheckFeature{}
+
+	return ebpfCheckFeat
+}
+
+type ebpfCheckFeature struct{}
+
+// ID returns the ID of the Feature
+func (f *ebpfCheckFeature) ID() feature.IDType {
+	return feature.EBPFCheckIDType
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *ebpfCheckFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	if dda.Spec.Features != nil && dda.Spec.Features.EBPFCheck != nil && apiutils.BoolValue(dda.Spec.Features.EBPFCheck.Enabled) {
+		reqComp.Agent = feature.RequiredComponent{
+			IsRequired: apiutils.NewBoolPointer(true),
+			Containers: []apicommonv1.AgentContainerName{apicommonv1.CoreAgentContainerName, apicommonv1.SystemProbeContainerName},
+		}
+	}
+
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *ebpfCheckFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
```

Comment:
Should we still add new features to datadogagent/v1alpha1? I'll check with the team on what our stance should be 

If we do decide to add the ebpf feature to v1alpha1, then I think we'd also need to add it to the conversion logic https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/v1alpha1/datadogagent_conversion_agent.go

Reply:
Let's not add the new feature to v1alpha1 since we GA'd the operator with v2alpha1. You can remove all of `ConfigureV1` and run `make generate` again.

Reply:
updated